### PR TITLE
migrate theta-module readout and related code to k4geo

### PR DESCRIPTION
**Pull request by giovannimarchiori**
Code changes related to k4geo migration of FCC segmentation classes and related utilities - this PR is basically a rebasing to the latest version of main of https://github.com/HEP-FCC/k4RecCalorimeter/pull/56

### CODE REVIEW

**Comment by giovannimarchiori** [comment]
Hi!
Just to understand - who's supposed to review this PR? @kjvbrt?
Thanks a lot, Giovanni

File: RecFCCeeCalorimeter/src/components/CaloTopoClusterFCCee.cpp [hunk 1]
```diff
@@ -69,11 +72,16 @@ StatusCode CaloTopoClusterFCCee::initialize() {
     error() << "Unable to retrieve ECal Barrel cell positions tool!!!" << endmsg;
     return StatusCode::FAILURE;
   }
-  // Check if cell position HCal Barrel tool available
-  if (!m_cellPositionsHCalBarrelTool.retrieve()) {
-    error() << "Unable to retrieve HCal Barrel cell positions tool!!!" << endmsg;
+  // Check if cell position HCal Barrel tool available (if name is set)
+  if (!m_cellPositionsHCalBarrelTool.empty()) {
```
Comment (kjvbrt) [inline RecFCCeeCalorimeter/src/components/CaloTopoClusterFCCee.cpp:76]: This condition seems to be always true, can you provide empty tool handle in the corresponding `declareProperty` above?

File: RecFCCeeCalorimeter/src/components/CaloTopoClusterFCCee.cpp [hunk 2]
```diff
@@ -69,11 +72,16 @@ StatusCode CaloTopoClusterFCCee::initialize() {
...
+    if (!m_cellPositionsHCalBarrelTool.retrieve()) {
+      error() << "Unable to retrieve HCal Barrel cell positions tool!!!" << endmsg;
+      return StatusCode::FAILURE;
+    }
+  }
+  if (!m_cellPositionsHCalBarrelNoSegTool.empty()) {
```
Comment (kjvbrt) [inline RecFCCeeCalorimeter/src/components/CaloTopoClusterFCCee.cpp:82]: Here as well

File: RecFCCeeCalorimeter/src/components/CaloTopoClusterFCCee.cpp (hunk 1 above)
Comment (giovannimarchiori) [reply RecFCCeeCalorimeter/src/components/CaloTopoClusterFCCee.cpp:77]: fixed in latest commit

**Comment by BrieucF** [comment]
Juraj had started to get rid of the position tools from topo clustering (assuming the cells have positioned attached to them): https://github.com/HEP-FCC/k4RecCalorimeter/pull/57 . Shall we do it here or adapt PR#57 after merging this PR?

**Comment by BrieucF** [comment]
https://github.com/key4hep/k4geo/pull/304 has been merged, we should merge this as well, I propose to deal with the cell position tools suppression in another PR

**Comment by kjvbrt** [comment]
I agree :)

**Comment by giovannimarchiori** [comment]
Thanks! And yes it's great to get rid of the positioning tools at clustering level since we can cluster cells that already have positions attached to them

**Pull request merged**